mypy/fastparse.py

```diff
@@ -394,13 +414,13 @@ def translate_module_id(self, id: str) -> str:
         return id
 
     def visit_Module(self, mod: ast3.Module) -> MypyFile:
+        self.type_ignore_comments = {ti.lineno for ti in mod.type_ignores}
         body = self.fix_function_overloads(self.translate_stmt_list(mod.body))
-        ignores = [ti.lineno for ti in mod.type_ignores]
-        ignores.extend(self.extra_type_ignores)
+        self.type_ignores.update({line: line for line in self.type_ignore_comments})
```

Comment:
I think it is less error-prone to reset `type_ignores` when visiting each file. I.e. here I would just use `self.type_ignores = {line: line for ...}`.